zerver/lib/events.py

```diff
@@ -628,6 +679,17 @@ def fetch_initial_state_data(
         state["can_create_web_public_streams"] = (
             realm.can_create_web_public_channel_group_id in settings_user_recursive_group_ids
         )
+
+        user_has_billing_access = (
+            realm.can_manage_billing_group_id in settings_user_recursive_group_ids
+        )
+        billing_info = get_billing_info(settings_user, user_has_billing_access)
+
+        state["show_billing"] = billing_info.show_billing
+        state["show_plans"] = billing_info.show_plans
+        state["show_remote_billing"] = billing_info.show_remote_billing
+        state["sponsorship_pending"] = billing_info.sponsorship_pending
```

Comment:
These should probably be namespaced in some way. Maybe it makes sense to just have a `realm_billing` section? Worth an "api design" discussion.